src/dataDisplay/tableHeaderCell.jsx

```diff
@@ -1,32 +1,360 @@
+        this.onClick = this.onClick.bind(this);
+        this.canCellKeepExpanding = this.canCellKeepExpanding.bind(this);
+        this.onDoubleClick = this.onDoubleClick.bind(this);
+        this.onResizeColumn = this.onResizeColumn.bind(this);
+        this.onResizeStop = this.onResizeStop.bind(this);
+        this.getResizingHandler = this.getResizingHandler.bind(this);
+        this.getMaxWidthPercentages = this.getMaxWidthPercentages.bind(this);
+    }
+
+    componentDidMount() {
+        const {
+            isResizable,
+        } = this.props;
+        if (isResizable) {
+            const currentWidth = _.get(this, 'header.current.clientWidth');
+            this.setState({
+                minWidth: currentWidth,
+            });
+        }
+        this.debouncedUpdateExpandedWidth = _.debounce(
+            (initialLoad) => {
+                const {
+                    shouldResetWhiteSpaceStyle,
+                    setResizableCellsWordWrapping,
+                    stickyTableContainerWidth,
+                    resizableColumnWidthPercentage,
+                    adjacentStickyColumnsTotalWidth,
+                    forceTableUpdate,
+                    stickyColumnCount,
+                } = this.props;
+                const {
+                    widthPercentage: currentWidthPercentage,
+                    minWidth,
+                } = this.state;
+                // eslint-disable-next-line max-len
+                const shouldUpdateExpandedWidth = isResizable && minWidth > 0 && shouldResetWhiteSpaceStyle;
+                if (shouldUpdateExpandedWidth) {
+                    // eslint-disable-next-line max-len
+                    const maxWidthPercentages = this.getMaxWidthPercentages(stickyTableContainerWidth);
+                    setResizableCellsWordWrapping(false);
+                    // This ensures the header will re adjust accordint to the content
+                    // in case the column and browser have been already resized
+                    this.header.current.style.width = '1%';
+                    this.header.current.style.minWidth = '';
+                    const updatedExpandedWidth = _.get(this, 'header.current.clientWidth', 0) + HEADER_CELL_WIDTH_OFFSET;
+                    // eslint-disable-next-line max-len
+                    const adjustedAdjacentPercentage = (adjacentStickyColumnsTotalWidth / stickyTableContainerWidth) * 100;
+                    const shouldDisableResizeHandler = stickyColumnCount > 1 &&
+                        // eslint-disable-next-line max-len
+                        updatedExpandedWidth + adjacentStickyColumnsTotalWidth > stickyTableContainerWidth * maxWidthPercentages[0];
+                    // eslint-disable-next-line max-len
+                    let expandedWidthPercentage = (updatedExpandedWidth / stickyTableContainerWidth) * 100;
+                    let widthPercentage;
+                    if (initialLoad) {
```

Comment:
Before and after `if(...) { ... }` are other places my preference is for blank lines:
```JavaScript
let widthPercentage;

if (initialLoad) {
    // ... do all the initial load things ...
}

// empty line, then whatever comes next ...
```

Reply:
Looks good.  Resolving.